autoremovetorrents/filter/path.py

```diff
@@ -0,0 +1,34 @@
+#-*- coding:utf-8 -*-
+
+from .filter import Filter
+
+class PathFilter(Filter):
+    def __init__(self, all_paths, ac, re):
+        '''
+        filter torrents by path (download directory)
+
+        Parameters:
+          all_paths (boolean): whether accept all paths
+          ac (list): accept paths
+          re (list): reject paths
+        '''
+        Filter.__init__(self, all_paths, ac, re)
+
+    def apply(self, torrents):
+        # Pick accepted torrents
+        accepts = set()
+        if self._all: # Accept all torrents (all_paths)
+            accepts = set(torrents)
+        elif len(self._accept) > 0: # Accept specific path torrents (paths)
+            for torrent in torrents:
+                path = torrent.download_dir
+                if any(map(path.startswith, self._accept)) or path == '':
```

Comment:
It's good, but it seems that it cannot work if we mix absolute paths and relative paths, for example, we set `path: ~/abc` and the path of `Torrent A` returned by our Transmission client is `/home/my/abc`.

I think this post can help us to solve this problem perfectly: https://stackoverflow.com/questions/3812849/how-to-check-whether-a-directory-is-a-sub-directory-of-another-directory.

Reply:
But the path may be in a remote host, I can't checkt it without ssh permission, it's would be a lot of change.

Reply:
You're right. Maybe we should not allow the user to use relative paths.